CONTRIBUTING.md

```diff
@@ -0,0 +1,71 @@
+This is mainly due to our bindings to other languages.
+
+Usually you should not run cargo update when creating a pull request even when adding/changing a dependency.
+Simply building the library should update the `Cargo.lock` file with the minimal changes needed.
+
+> [!NOTE]
+> This is not normal / best practice for most libraries.
+> If you are writing your own library, your `Cargo.lock` file is not used when the library is included in another project.
+> This means your CI will be testing with an out of date set of dependencies compared to your users, missing potential problems.
+> See the [official documentation](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries) for more information. 
+
+### CHANGELOG.md
+
+Our CHANGELOG.md files are auto generated using [Release Please](https://github.com/googleapis/release-please) and should not be edited manually.
+
+To control what is in the CHANGELOG.md for your change, use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) when writing your commit messages.
+
+#### Example Commit Messages
+Taken from [Conventional Commits documentation](https://www.conventionalcommits.org/en/v1.0.0/#summary):
+> feat: allow provided config object to extend other configs
+>
+> BREAKING CHANGE: `extends` key in config file is now used for extending other config files
+
+> fix: prevent racing of requests
+>
+> Introduce a request id and a reference to latest request. Dismiss
+incoming responses other than from latest request.
+
+### Testing Locally
+
+We have some platform specific tests that are not run with `cargo test` from the project root by default.
+
+1. Run test that work on all platforms
+   ``` shell
+   cargo test
+   ```
+2. Run platform specific tests.
+
+   Run the appropriate command for your platform
+   ``` shell
+   cargo test -p accesskit_windows
+   cargo test -p accesskit_unix
+   cargo test -p accesskit_macos
+   ```
+   Not all platforms have tests at this time, but they may in the future.
+
+> [!WARNING]
+> **Windows**: Some tests may fail if the created window looses focus while testing. This can also happen when using the terminal built into your IDE. Try running them from your native terminal if you get failures. 
+
+### Common issues
+
+#### CI Errors
+
+If you get an error in the CI step `Run cmake --build build` that includes near the end of the error: 
+```
+ERROR: Couldn't generate bindings for /home/runner/work/accesskit/accesskit/bindings/c.
+```
+Somewhere in the big block of text in the error message it also says:
+```
+unknown feature `proc_macro_span_shrink`
+```
+
+This is likely not an issue with your PR and just means `proc-macro2` needs to be updated in our Cargo.lock file. 
+You probably want to make this change in a separate PR as `proc-macro2` can update often and your PR could become out of date if it takes a while to be reviewed.
```

Comment:
Just lmk if you think this isn't a good place for this, was on the fence about including this myself. Not sure how often it actually happens etc.

Reply:
I don't think that exact breakage would repeat. It would perhaps be something different (or perhaps not).

Reply:
Should I just take this section out, or do you think it's worth having, just need to make it not specific to the `proc_macro_span_shrink` feature? 

Could also take it out for now and consider adding something if it keeps being a problem in the future.

Reply:
Removed that section for now to keep things moving. Might be better in a future PR if it keeps being a problem. (or feel free to suggest a revision of it)